### ISSUE TITLE
last issue -- questions about the phasemap output

### ISSUE THREAD

**Issue by wheaton5** [thegenemyers/PHASE-MERS#4]
Ok so here are some examples

15906   3080264 c
        15924   134873123       %
        15963   79233114        g
        16082   92274759        %
        16103   236847240       t
        16105   266141784       a
        16123   115277944       g
        16143   113246232       a
        16160   30081118        c
        16246   13828099        %
        16263   64421946        -
        16264   262602883       s
        16268   193003656       t
        16281   2228246 -
        16282   54067255        g
        16284   219217962       :
        16289   4784178 a
        16293   220790910       -
        16324   219217990       :

You said previously the - are the homozygous, so what are : and %?

The positions are of the start of the kmer?

how do you get two kmer pairs which are overlapping one another
 16103   236847240       t
        16105   266141784       a
?

and how do you get a homozygous kmer 1 base before a het pair
 16263   64421946        -
        16264   262602883       s
well presumably both of these signals could happen if the het pair actually came from paralogous sequence instead of heterozygosity. do you agree these are the only ways this could arise? OR I guess the hom kmer might not be a hom kmer in this last case. But the previous case i don't think i see how it could happen (2 overlapping het kmers).

Best,
Haynes

And after this i'll stop taking up your time. You have been very generous in all of this and of course we will include you in papers as this was done custom for me.

**Comment by wheaton5** [comment]
Looking at your code these should not happen and should be a result of the 3 lower bits (code) to be 5-7 and you reading off the end of the DNA array. I'm rerunning from the start to make sure it wasn't something I did wrong (still could be but anyway).

**Comment by thegenemyers** [comment]
Yes, the : and % are troublesome, something is amiss.  I believe the 
overlapping
hets were due to palindromes and the only requirment for a hom-mer is 
that it
not be at the same place as a het-mer, so as long as its count is in 
range it
can happend.  But the : and % are troublesome.

I'm about to get back to modifying/completing Phasemap as we discussed 
so it gives
orientation and barcodes for reads.

This last I'm a bit confused about -- are you really going to sort all 
the bar codes after
the fact?  In my conception the .fastq files should themselves be sorted 
by bar code :-)

   -- Gene

**Comment by wheaton5** [comment]
All I need is to associate the barcodes with the reads. I don't need to sort them. When I load them in there is a hashmap from barcode id (an int, not the barcode itself) to a list of kmer ids and also a hashmap from kmer_id to a list of barcode ids.

**Comment by wheaton5** [comment]
But we do need to figure out the %, :, s stuff here (it is due to the code being > 4, but not sure how that is possible). Could that happen if I'm creating the ktab from the short reads but am creating the profiles on the ccs reads? Doesn't seem likely.
